### PR TITLE
Added GET /api/articles/:articleId endpoint

### PR DESCRIPTION
**Pull request by wakenado**

### CODE REVIEW

File: controllers/app.controllers.js [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+  const newsTopics = await fetchNewsTopics();
+  res.status(200).send({ topics: newsTopics });
+};
+
+exports.getArticleById = async (req, res, next) => {
+  try {
+    const { articleId } = req.params;
+    const response = await fetchArticleById(articleId);
+    if (response.length === 0) {
+      res.status(404).send({ message: "404 - Not Found" });
```
Comment (Angaca) [inline controllers/app.controllers.js:14]: You should promise reject here and write a custom error handling function to catch this custom error

File: controllers/app.controllers.js [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+    } else {
+      res.status(200).send({ message: response });
```
Comment (Angaca) [inline controllers/app.controllers.js:16]: needds to change accordingly

File: __tests__/app.test.js [hunk 3]
```diff
@@ -27,15 +27,40 @@ describe("app", () => {
...
+      expect(status).toBe(200);
     });
-    test("should return 404 error message", async () => {
-      const response = await request(app).get("/api/unknown");
-      expect(response.body.message).toBe("404 Path Not Found");
+    test("should return an article object", async () => {
+      const response = await request(app).get("/api/articles/1");
+      expect(response.body).toBeInstanceOf(Object);
+      expect(response.body.message).toHaveLength(1);
+      expect(response.body.message[0]).toEqual({
```
Comment (Angaca) [inline __tests__/app.test.js:45]: it shouldn't be in a key of message and when we send a single element it shouldn't be in an array

**Review by Angaca — CHANGES_REQUESTED** [review]
You need to amend few details in your code, please see my comments

File: controllers/app.controllers.js (hunk 1 above)
Comment (wakenado) [reply controllers/app.controllers.js:14]: Done

File: __tests__/app.test.js (hunk 3 above)
Comment (wakenado) [reply __tests__/app.test.js:45]: Done

**Review by Angaca — CHANGES_REQUESTED** [review]
Almost there, we should't attach any status to our normal responses. Any status assertion in the test should go down request(app).expect(200) (or any other status code you expect). Moreover anything you send should be in a relevant key. Article should be in a key of article and not being free in the body

File: controllers/app.controllers.js [hunk 4]
```diff
@@ -1,6 +1,21 @@
-const { fetchNewsTopics } = require("../models/app.models");
+const res = require("express/lib/response");
+const { fetchNewsTopics, fetchArticleById } = require("../models/app.models");
+const { checkArticleExist } = require("../models/articles.models");
 
 exports.getNewsTopics = async (req, res) => {
-  const result = await fetchNewsTopics();
-  res.status(200).send({ topics: result });
+  const newsTopics = await fetchNewsTopics();
+  res.status(200).send({ topics: newsTopics });
```
Comment (Angaca) [inline controllers/app.controllers.js:7]: can rename newTopics to topics so you can only send {topics}

Comment (wakenado) [reply controllers/app.controllers.js]: Done

**Pull request merged**